### PR TITLE
Refactor Workers

### PR DESCRIPTION
**Pull request by tjholm**
Refactor workers to better separate interfaces. This will allow for easier testing as well as better separation of protocol handling vs. worker types e.g. A subscription worker could have a gRPC handler or a HTTP handler.

### CODE REVIEW

**Review by jyecusch — APPROVED** [review]
I like the separation, but I think we could improve the naming for these things. handlers/worker feel overloaded and hard to know which is which without digging into the code.

The handler is more like a receiver/adapter for streams/events/etc. and the worker actually handles/does the work triggered/requested by the events.

**Comment by tjholm** [comment]
> I like the separation, but I think we could improve the naming for these things. handlers/worker feel overloaded and hard to know which is which without digging into the code.
> 
> The handler is more like a receiver/adapter for streams/events/etc. and the worker actually handles/does the work triggered/requested by the events.

Will refactor this to rename to adapter

**Review by HomelessDinosaur — CHANGES_REQUESTED** [review]

**Comment by HomelessDinosaur** [comment]
Just those small changes, otherwise looks good

File: pkg/worker/route_worker.go [hunk 1]
```diff
@@ -107,11 +108,11 @@ type RouteWorkerOptions struct {
 
 // Package private method
 // Only a pool may create a new faas worker
-func NewRouteWorker(stream pb.FaasService_TriggerStreamServer, opts *RouteWorkerOptions) *RouteWorker {
+func NewRouteWorker(handler Adapter, opts *RouteWorkerOptions) *RouteWorker {
```
[the remark this answers is not in the document]
Comment (tjholm) [reply pkg/worker/route_worker.go]: done

File: pkg/worker/faas_worker.go [hunk 2]
```diff
@@ -14,19 +14,27 @@
...
+	return true
+}
+
+func (s *FaasWorker) HandlesEvent(trigger *triggers.Event) bool {
+	return true
 }
 
 // NewFaasWorker - Create a new FaaS worker
-func NewFaasWorker(stream pb.FaasService_TriggerStreamServer) *FaasWorker {
+func NewFaasWorker(handler Adapter) *FaasWorker {
```
[the remark this answers is not in the document]
Comment (tjholm) [reply pkg/worker/faas_worker.go]: done

**Review by HomelessDinosaur — APPROVED** [review]

**Pull request merged**